how-to/containerd-kata.md

```diff
@@ -0,0 +1,345 @@
+
+And check if `containerd` is now available
+  
+```bash
+> command -v containerd
+```
+
+### Install CNI plugins
+
+> **Note:** You don't need install CNI plugins if you don't want to use containerd with Kubernetes.
+> And if you have installed Kubernetes with `kubeadm`, you may have already had CNI plugins installed.
+
+You may manually install CNI plugins as follows
+
+```bash
+> go get github.com/containernetworking/plugins
+> pushd $GOPATH/src/github.com/containernetworking/plugins
+> ./build_linux.sh
+> mkdir /opt/cni
+> cp -r bin /opt/cni/
+> popd
+```
+
+### Install cri-tools
+
+> **Note:** `cri-tools` is a set of tools for CRI for the development and testing. For users only want to use containerd
+> with kubernetes, you may skip the `cri-tools`.
+
+You may install the `cri-tools` from source code:
+
+```bash
+> go get github.com/kubernetes-incubator/cri-tools
+> pushd $GOPATH/src/github.com/kubernetes-incubator/cri-tools
+> make
+> make install
+> popd
+```
+
+## Configuration
+
+### Configure containerd to use Kata Containers
+
+By default, the configuration of containerd locates at `/etc/containerd/config.toml`, and the 
+`cri` plugins are placed in the following section.
+
+```toml
+[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.default_runtime]
+        #runtime_type = "io.containerd.runtime.v1.linux"
+
+    [plugins.cri.cni]
+      # conf_dir is the directory in which the admin places a CNI conf.
+      conf_dir = "/etc/cni/net.d"
+```
+
+Let's add Kata Containers to the configurations.
+
+#### Kata Containers as a RuntimeClass
+
+For 
+- Kata Containers v1.5.0 or above (including 1.5.0-rc)
+- Containerd v1.2.0 or above
+- Kubernetes v1.12.0 or above
+
+The  RuntimeClass is suggested.
+
+The following configuration include three runtime classes:
+- `plugins.cri.containerd.runtimes.runc`: the runC, and it is the default runtime;
+- `plugins.cri.containerd.runtimes.kata`: the dot connected string `io.containerd.kata.v2` will be translated to `containerd-shim-kata-v2`, which is the kata implementation of shim v2 API
+- `plugins.cri.containerd.runtimes.katacli`: the `containerd-shim-runc-v1` will call `kata-runtime`, which is the legacy way deprecated by the `containerd-shim-kata-v2`
+
+```toml
+    [plugins.cri.containerd]
+      no_pivot = false
+    [plugins.cri.containerd.runtimes]
+      [plugins.cri.containerd.runtimes.runc]
+         runtime_type = "io.containerd.runc.v1"
+         [plugins.cri.containerd.runtimes.runc.options]
+           NoPivotRoot = false
+           NoNewKeyring = false
+           ShimCgroup = ""
+           IoUid = 0
+           IoGid = 0
+           BinaryName = "runc"
+           Root = ""
+           CriuPath = ""
+           SystemdCgroup = false
+      [plugins.cri.containerd.runtimes.kata]
+         runtime_type = "io.containerd.kata.v2"
+         [plugins.cri.containerd.runtimes.kata.options]
```

Comment:
By now containerd hasn't supported any kata options, so I think it's better to remove them. 
we had a PR https://github.com/containerd/containerd/pull/2916 to track on how to pass kata options from containerd to shimv2.